src/main/java/com/direwolf20/mininggadgets/common/gadget/MiningCollect.java

```diff
@@ -68,6 +73,10 @@ private static boolean isValid(BlockPos pos, World world) {
         if (state.getBlockHardness(world, pos) < 0)
             return false;
 
+        // Reject, if the dimension blocks the player from mining the block
+        if(!world.getDimension().canMineBlock(player, pos))
```

Comment:
Can you put this at the top so it's rejected instantly. I'd prefer to do this before state checking.

Reply:
Moved it up, as far as possible. Still the state checking for the RenderBlock is necessary to prevent it from being rejected by the `canMineBlock` function.